### ISSUE TITLE
self.context is None in VeoProc.__dealloc__

### ISSUE THREAD

**Issue by FreddieWitherden** [SX-Aurora/py-veo#3]
Upon termination I sometimes encounter the error:
```
TypeError: object of type 'NoneType' has no len()
Exception ignored in: 'veo._veo.VeoProc.__dealloc__'
TypeError: object of type 'NoneType' has no len()
```
which implies that `VecProc.context` is somehow getting set to None before `__dealloc__` is called.  Looking through the source I am not entirely sure how this can occur as we do not appear to assign to context anywhere (although Cython may be doing something behind the scenes that I am not following).

**Comment by FreddieWitherden** [comment]
Just as a followup the Cython documentation seems to imply that by the time `__dealloc__` is called that the object may be in a partially deconstructed state already

> You need to be careful what you do in a __dealloc__() method. By the time your __dealloc__() method is called, the object may already have been partially destroyed and may not be in a valid state as far as Python is concerned, so you should avoid invoking any Python operations which might touch the object. In particular, don’t call any other methods of the object or do anything which might cause the object to be resurrected. It’s best if you stick to just deallocating C data.

(source: https://cython.readthedocs.io/en/latest/src/userguide/special_methods.html#finalization-method-dealloc)

**Comment by efocht** [comment]
Thanks for the hint and for finding this! I'll remove the deletion of the contexts and only leave the `veo_proc_destroy()` call there.

**Comment by efocht** [comment]
Included in v1.4.0 release/tag. Closing.
